internal/exporter/pollingexporter/polling_exporter.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pollingexporter
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/jaegertracing/jaeger/model"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	jaegertranslator "go.opentelemetry.io/collector/translator/trace/jaeger"
+)
+
+var marshaler = &jsonpb.Marshaler{}
+
+// pollingExporter ...
+type pollingExporter struct {
+	endpoint string
+
+	ch      chan *model.Batch
+	clients []*client
+	mu      sync.Mutex
+}
+
+func (e *pollingExporter) ConsumeTraces(_ context.Context, td pdata.Traces) error {
+	batches, err := jaegertranslator.InternalTracesToJaegerProto(td)
+	if err != nil {
+		return err
+	}
+	for _, batch := range batches {
+		go func(b *model.Batch) {
+			e.ch <- b
+		}(batch)
+	}
+	return nil
+}
+
+func (e *pollingExporter) handler(w http.ResponseWriter, r *http.Request) {
+	opts, err := parseOptions(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	c := newClient(opts)
+
+	e.mu.Lock()
+	e.clients = append(e.clients, c)
+	e.mu.Unlock()
+
+	result, err := c.response()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
404 makes more sense than 500 to me, so maybe response should also include a status code?

Reply:
done